### DISCUSSION TITLE
Attribute [mailPreview] does not exist.

### DISCUSSION THREAD

**Discussion by BennyAlex** [themsaid/laravel-mail-preview#67] [Q&A]
Hello,

I added `Route::mailPreview()` inside routes/web.php and getting the following error:

>   Attribute [mailPreview] does not exist.
> 
>   at vendor/laravel/framework/src/Illuminate/Routing/RouteRegistrar.php:93

I already tried clearing the cache.

PHP 7.3.25
laravel 8.27

**Comment by m-bosch** [comment]
Same issue here and reported with more info on #68. for me it looks like it has to do something with the `php artisan clear-compiled` function i run in my deploy scripts.

**Reply by m-bosch** [reply]
Because i am not the only one perhaps @freekmurze has (almost always) a solution.

**Reply by BennyAlex** [reply]
@freekmurze can you explain, wh do you close the issues? These is a real bug, so it has its right to stay in issues

**Reply by freekmurze** [reply]
The tests prove that the package is working. Also, in my projects it is working well.

So, I consider this to be a bug somewhere in your code, and not in the package.

The issue tracker is only for bugs inside the page.

Should you find that the package is in error, submit a PR to fix it.

**Comment by m-bosch** [comment]
My problem was solved by checking online php version als CLI version. these were different because of that the wrong package was installed.

**Comment by BennyAlex** [comment]
@freekmurze 
hey, Looks like the installation with composer gets broken? 
![image](https://user-images.githubusercontent.com/17402724/112126481-c8952300-8bc4-11eb-94bc-27a6a435e554.png)

there is only the src folder, when I try to include "\Spatie\MailPreview\Http\Middleware\AddMailPreviewOverlayToResponse::class," it can't be found.

I already reinstalled it, after upgrading to composer v2. Still only src folder. Maybe your upload is broken? Or which php version do you support? I am using PHP 7.4.3

**Comment by m-bosch** [comment]
We solved it to remove the route, and register it tlike below in the appserviceprovider

```if ($this->app->environment('local')) {
            $kernel->appendMiddlewareToGroup('web', \Spatie\MailPreview\Http\Middleware\AddMailPreviewOverlayToResponse::class);
        }```


also in the route serviceprovider we added some code for the frontendroutes:

if ($this->app->environment('local')) {
                Route::mailPreview('digifactory-mail-preview');
            }


this way it works fine for us.

**Reply by BennyAlex** [reply]
thanks for response. Can you show the content of your ventor/spartie/MailPreview folder pls? I wonder if you have an http folder there

**Reply by r-pennings** [reply]
Not sure if you have already found a solution for this, but this is mine:
In order to make it work, you will have to run `composer require spatie/laravel-mail-preview` while PHP is running at version `^8.0` locally. When PHP is running on version 7, it will not install the package properly

**Comment by BrendanTWhite** [comment]
For me, I had installed the package with `composer require --dev` which meant that it worked fine in local, but when I migrated to my staging environment, I got the `Attribute [mailPreview] does not exist` error.

This is because I had `Route::mailPreview();` in my `routes/web.php` file, and `mailPreview` doesn't exist in Staging if you installed this package with `composer require --dev `.

The quickest fix for this problem is to move this package from `"require-dev"` to `"require"` in `composer.json` and then run `composer install` again.
